### PR TITLE
Accounting Style Renaming: proper manual migration

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

ADP-523

# Overview

- 016f60caf7d8efcd72686665ebd88c05d1230d2d
  :round_pushpin: **reproduce deserialization issue with regards to account style change**
    I originally added a 'backward-compatible' function step to the deserialization function of value of type 'AccountingStyle'. The problem is that it only concerns objects that are being deserialized. Any 'AccountingStyle' value that is queried will inevitably fail because columns in an existing database would still be using the old serialization format. As a consequence, fetching all addresses of an existing sequential wallet returns an empty set, regardless of what is in the database.

- 0a6c36c919a2eadaa70f5770feceddeec100f7b6
  :round_pushpin: **use a proper manual database migration for renaming AccountingStyle   instead of the hacky in-place adjustment in the deserializer.**

# Comments

Whoops...

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs [hunk 1]
```diff
@@ -262,6 +263,10 @@ spec = do
                 , minBound
                 )
 
+        it "'migrate' db with old text serialization for 'AccountingStyle'" $
+            testMigrationAccountingStyle @ShelleyKey
+                "shelleyAccountingStyle-v2020-10-13.sqlite"
```
Comment (piotr-iohk) [inline lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs:268]: :1st_place_medal:

**Review by piotr-iohk — APPROVED** [review]
lgtm. 

Works on testnet!

**Comment by piotr-iohk** [comment]
Actually creating new wallet here results in:
```
[cardano-wallet.api-server:Info:156] [2020-10-22 13:49:42.22 UTC] [RequestId 1] [POST] /v2/wallets
[cardano-wallet.wallet-engine:Error:157] [2020-10-22 13:49:42.24 UTC] Worker has exited unexpectedly: SQLite3 returned ErrorError while attempting to perform prepare "UPDATE seq_state_address SET accounting_style = \"utxo_internal\" WHERE accounting_style = \"u_tx_o_internal\"": no such table: seq_state_address
[cardano-wallet.api-server:Error:156] [2020-10-22 13:49:42.24 UTC] [RequestId 1] 500 Internal Server Error in 0.019850024s
```
(Seems that tests complain about that also)

**Comment by KtorZ** [comment]
@piotr-iohk yes, and same for random wallet, see the last change here: https://github.com/input-output-hk/cardano-wallet/compare/a74532de74f0c0e25e3b45ba73b89d1e3eaf0a7f..7257a4cee4d363f67671adfb11698d7049a5d115

We need to control that the table exists first :sweat_smile:  ...

**Comment by KtorZ** [comment]
bors merge

**Comment by KtorZ** [comment]
bors merge

**Comment by KtorZ** [comment]
bors merge

**Comment by KtorZ** [comment]
bors merge

**Pull request merged**